docs_src/microservices/device/supported/device-onvif-camera/Walkthrough/custom-build.md

```diff
@@ -0,0 +1,190 @@
+   ```
+
+1. Update `.env` file to add the registry and image version variable for device-onvif-camera:
+      Add the following registry and version information:
+      ```env
+      DEVICE_ONVIFCAM_VERSION=0.0.0-dev
+      ```
+
+1. Update the `add-device-onvif-camera.yml` to point to the local image.
+      ```yml
+      services:
+         device-onvif-camera:
+            image: edgexfoundry/device-onvif-camera:${DEVICE_ONVIFCAM_VERSION}
+      ```
+
+## Additional Configuration
+
+Here is some information on how to specially configure parts of the service beyond the provided defaults.  
+
+### Configure the Device Profiles
+
+The device profile contains general information about the camera and includes all of the device resources and commands that the device resources can use to manage the cameras. The default profile found at `cmd/res/devices/camera.yaml` contains all possible resources a camera could implement. Enable and disable supported resources in this file, or create an entirely new profile. It is important to set up the device profile to match the capabilities of the camera. Information on the resources supported by specific cameras can be found [here](../supplementary-info/ONVIF-protocol.md#tested-onvif-cameras). Learn more about device profiles in EdgeX [here.](https://docs.edgexfoundry.org/1.2/microservices/device/profile/Ch-DeviceProfile/)
+
+```yaml
+name: "onvif-camera" # general information about the profile
+manufacturer:  "Generic"
+model: "Generic ONVIF"
+labels:
+  - "onvif"
+description: "EdgeX device profile for ONVIF-compliant IP camera."
+
+deviceResources:
+  # Network Configuration
+  - name: "Hostname" # an example of a resource with get/set values
+    isHidden: false
+    description: "Camera Hostname"
+    attributes:
+      service: "Device"
+      getFunction: "GetHostname"
+      setFunction: "SetHostname"
+    properties:
+      valueType: "Object"
+      readWrite: "RW"
+```
+<p align="left">
+   <i>Sample: Snippet from camera.yaml</i>
+</p>
+
+
+### Configure the Provision Watchers
+
+The provision watcher sets up parameters for EdgeX to automatically add devices to core-metadata. They can be configured to look for certain features, as well as block features. The default provision watcher is sufficient unless you plan on having multiple different cameras with different profiles and resources. Learn more about provision watchers [here](https://docs.edgexfoundry.org/2.2/microservices/core/metadata/Ch-Metadata/#provision-watcher).
+
+```json
+{
+    "name":"Generic-Onvif-Provision-Watcher",
+    "identifiers":{  // Use the identifiers to filter through specific features of the protocol
+         "Address": ".",
+         "Manufacturer": "Intel", // example of a feature to allow through 
+         "Model": "DFI6256TE" 
+    },
+    "blockingIdentifiers":{
+    },
+    "serviceName": "device-onvif-camera",
+    "profileName": "onvif-camera",
+    "adminState":"UNLOCKED"
+}
```

Comment:
need to make sure this is still valid, as i believe they have changed the format recently